### PR TITLE
Upgrade Zeebe to 8.2.0

### PR DESCRIPTION
**Pull request by oleschoenburg**
This should make zdb compatible with  8.2.0:

1. Various moves and renames of Zeebe internal classes
2. Enabling of RocksDB WAL in various tests. Otherwise, the rocksdb instance opened in tests doesn't see all writes. This was new to me. Maybe this impacts not just tests since you can point zdb to a runtime dir where the same could happen?
3. Updates expected log indices. I didn't check in detail but probably changed because of batch processing?
4. Updates expected state keys. I think this changed because of the stream platform & engine split?
5. Provides a No-Op implementation of `MetaStore` which I think is not used by zdb.

### CODE REVIEW

**Comment by Zelldon** [comment]
:D Thanks @oleschoenburg :rocket: I was actually planning to work on something related yesterday. I want to continue with https://github.com/Zelldon/zdb/pull/243 so we remove the dependency to state and zb-db classes, which should remove the burden of upgrades and breaking api's. 

But your PR is totally fine, then I will do this a bit later I guess.

File: backend/src/main/kotlin/io/zell/zdb/log/ApplicationRecord.kt [hunk 1]
```diff
@@ -15,8 +14,7 @@ class ApplicationRecord(val index: Long, val term : Long) : PersistedRecord {
     }
 
     override fun toString(): String {
-
-        val entriesJson = entries.map { it.toJson() }.joinToString()
+        val entriesJson = entries.map { "\"${it.toString().replace("\"", "'")}\"" }.joinToString()
```
Comment (Zelldon) [inline backend/src/main/kotlin/io/zell/zdb/log/ApplicationRecord.kt:17]: :question: Why this change? :thinking: toJson should still work?

File: backend/src/main/kotlin/io/zell/zdb/log/LogSearch.kt [hunk 2]
```diff
@@ -23,7 +24,7 @@ class LogSearch (logPath: Path) {
             val entry = reader.next()
 
             if (entry.isApplicationEntry) {
-                val applicationEntry = entry.applicationEntry
+                val applicationEntry = entry.applicationEntry as SerializedApplicationEntry
```
Comment (Zelldon) [inline backend/src/main/kotlin/io/zell/zdb/log/LogSearch.kt:27]: :question: Did this changed as well ? :thinking:

File: backend/src/main/kotlin/io/zell/zdb/state/blacklist/BlacklistState.kt [hunk 3]
```diff
@@ -16,7 +15,8 @@ class BlacklistState(private var readonlyDb: ReadonlyTransactionDb) {
 
         val processInstanceKey = DbLong()
         val blackListColumnFamily =
-            readonlyDb.createColumnFamily(ZbColumnFamilies.BLACKLIST,
+            readonlyDb.createColumnFamily(
+                ZbColumnFamilies.BLACKLIST,
```
Comment (Zelldon) [inline backend/src/main/kotlin/io/zell/zdb/state/blacklist/BlacklistState.kt:19]: :question: Do you use google format? I think I need to configure the project settings in the repo, and reformat the code. I actually have this on my todo list :sweat_smile:

File: backend/src/main/kotlin/io/zell/zdb/state/general/GeneralState.kt [hunk 4]
```diff
@@ -4,24 +4,24 @@ import io.camunda.zeebe.db.impl.DbCompositeKey
...
+import io.camunda.zeebe.stream.impl.state.DbLastProcessedPositionState
 import io.zell.zdb.db.readonly.transaction.ReadonlyTransactionDb
 import io.zell.zdb.state.incident.IncidentState
 import io.zell.zdb.state.instance.InstanceState
 import java.nio.file.Path
 
 class GeneralState(statePath: Path) {
 
-    private var zeebeDbState: ZeebeDbState
+    private var zeebeDbState: ProcessingState
```
Comment (Zelldon) [inline backend/src/main/kotlin/io/zell/zdb/state/general/GeneralState.kt:19]: Ah true, I guess I need to fix some stuff in eze as well :sweat_smile:

File: backend/src/test/kotlin/LogHelperTest.java [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
 import io.camunda.zeebe.logstreams.impl.log.LoggedEventImpl;
+import io.camunda.zeebe.logstreams.impl.serializer.DataFrameDescriptor;
 import io.camunda.zeebe.protocol.Protocol;
 import io.camunda.zeebe.protocol.impl.record.RecordMetadata;
 import io.camunda.zeebe.protocol.record.ValueType;
 import io.zell.zdb.log.LogHelperKt;
 import org.agrona.ExpandableArrayBuffer;
 import org.junit.jupiter.api.Test;
 
+import static org.assertj.core.api.Assertions.assertThat;
```
Comment (Zelldon) [inline backend/src/test/kotlin/LogHelperTest.java:11]: :thinking: same as above

File: backend/src/test/kotlin/ZeebeBlacklistStateTest.java [hunk 6]
```diff
@@ -57,6 +45,7 @@ public class ZeebeBlacklistStateTest {
 
   @Container
   public static ZeebeContainer zeebeContainer = new ZeebeContainer()
+      .withEnv("ZEEBE_BROKER_EXPERIMENTAL_ROCKSDB_DISABLEWAL", "false")
```
Comment (Zelldon) [inline backend/src/test/kotlin/ZeebeBlacklistStateTest.java:48]: :question: What happened when you not add this? The test failed? Interesting.

File: backend/src/test/kotlin/ZeebeLogTest.java [hunk 7]
```diff
@@ -195,7 +195,7 @@ public void shouldBuildLogContent() throws JsonProcessingException {
     final var content = logContentReader.content();
 
     // then
-    assertThat(content.getRecords()).hasSize(24);
+    assertThat(content.getRecords()).hasSize(13);
```
Comment (Zelldon) [inline backend/src/test/kotlin/ZeebeLogTest.java:198]: :question: Ok this is unexpected. Can we take a look at the log :thinking:

File: backend/src/test/kotlin/ZeebeStateTest.java [hunk 8]
```diff
@@ -137,7 +139,7 @@ public void shouldVisitValuesAsJson() {
     experimental.visitDBWithJsonValues(jsonVisitor);
 
     // then
-    assertThat(incidentMap).containsValue("{\"incidentRecord\":{\"errorType\":\"IO_MAPPING_ERROR\",\"errorMessage\":\"failed to evaluate expression '{bar:foo}': no variable found for name 'foo'\",\"bpmnProcessId\":\"process\",\"processDefinitionKey\":2251799813685249,\"processInstanceKey\":2251799813685251,\"elementId\":\"incidentTask\",\"elementInstanceKey\":2251799813685262,\"jobKey\":-1,\"variableScopeKey\":2251799813685262}}");
+    assertThat(incidentMap).containsValue("{\"incidentRecord\":{\"errorType\":\"IO_MAPPING_ERROR\",\"errorMessage\":\"failed to evaluate expression '{bar:foo}': no variable found for name 'foo'\",\"bpmnProcessId\":\"process\",\"processDefinitionKey\":2251799813685249,\"processInstanceKey\":2251799813685251,\"elementId\":\"incidentTask\",\"elementInstanceKey\":2251799813685260,\"jobKey\":-1,\"variableScopeKey\":2251799813685260}}");
```
Comment (Zelldon) [inline backend/src/test/kotlin/ZeebeStateTest.java:142]: Yeah this is a bit unfortunate, looks like the ordering changes

**Review by Zelldon — COMMENTED** [review]
Hey @oleschoenburg I had some questions and comment can you take a look before we merge it ? :)

File: backend/src/main/kotlin/io/zell/zdb/log/ApplicationRecord.kt (hunk 1 above)
Comment (oleschoenburg) [reply backend/src/main/kotlin/io/zell/zdb/log/ApplicationRecord.kt:17]: Oops, that's my bad. It was left over when I tried to get it running on 8.2 logs without actually doing the upgrade. Good catch :+1:

File: backend/src/main/kotlin/io/zell/zdb/log/LogSearch.kt (hunk 2 above)
Comment (oleschoenburg) [reply backend/src/main/kotlin/io/zell/zdb/log/LogSearch.kt:27]: Yes, there's serialized and unserialized. Since we are only reading from the log, log entries are serialized.

File: backend/src/main/kotlin/io/zell/zdb/state/blacklist/BlacklistState.kt (hunk 3 above)
Comment (oleschoenburg) [reply backend/src/main/kotlin/io/zell/zdb/state/blacklist/BlacklistState.kt:19]: I don't think I enabled google-java-format but still my IntelliJ had very different _opinions_ on formatting :sweat_smile:. I tried to keep formatting changes to a minimum though.

File: backend/src/test/kotlin/ZeebeBlacklistStateTest.java (hunk 6 above)
Comment (oleschoenburg) [reply backend/src/test/kotlin/ZeebeBlacklistStateTest.java:48]: Yes, otherwise the test could not read anything from rocksdb. As I wrote in the PR description, this was surprising to me as well.

File: backend/src/test/kotlin/ZeebeLogTest.java (hunk 7 above)
Comment (oleschoenburg) [reply backend/src/test/kotlin/ZeebeLogTest.java:198]: Is it? It's the amount of journal records that went from 24 to 13. I suspect the number of Zeebe records remained the same.

File: backend/src/main/kotlin/io/zell/zdb/log/ApplicationRecord.kt (hunk 1 above)
Comment (oleschoenburg) [reply backend/src/main/kotlin/io/zell/zdb/log/ApplicationRecord.kt]: I reverted this change

File: backend/src/test/kotlin/ZeebeLogTest.java (hunk 7 above)
Comment (Zelldon) [reply backend/src/test/kotlin/ZeebeLogTest.java:198]: AH ok then I misinterpreted that sorry.

**Comment by oleschoenburg** [comment]
Pinging @Zelldon as this is ready to merge I think? :heart:

**Comment by Zelldon** [comment]
I actually wanted to ask you in the morning that the PR diff doesn't show an version update of Zeebe, which confused me. Is this expected?

**Comment by oleschoenburg** [comment]
Ah right, I only had to remove the _override_ of the zeebe version used for the backend here: https://github.com/Zelldon/zdb/pull/252/files#diff-34049c3bc6deee4bbf269544338e0450399140da63fec684096d1ae0ce70b4bbL32

It was already set to 8.2 in the parent pom.

**Comment by Zelldon** [comment]
Thanks again @oleschoenburg :rocket:
